### ISSUE TITLE
boost 

### ISSUE THREAD

**Issue by flourinekk** [cpp-pm/hunter#750]
Boost program options doesn't compile.

**Comment by NeroBurner** [comment]
Which compiler are you using. If it is VS2022 17.10 then we need to wait for the next boost release

https://github.com/boostorg/boost/issues/914

**Comment by flourinekk** [comment]
Yeah, that's the problem then, how i can edit hunter files to use workaround so it doesn't replace me files?

**Comment by NeroBurner** [comment]
If there is a new release you can point your URL to the new package to download using a `cmake/hunter.config` file. Until then the only workaround is to use an older MSVC version with vc143 runtime instead of vc144

**Comment by Deneb128** [comment]
> If there is a new release you can point your URL to the new package to download using a `cmake/hunter.config` file. Until then the only workaround is to use an older MSVC version with vc143 runtime instead of vc144

Problem is not even vc144. I don't even know how but in two days difference with no update of Visual Studio, hunter cannot recognize `MSVC_VERSION: 1940`
Forcing it to 1939 fails to compile, but if I try to do it manually inside the hunter folder with: `.\b2 variant=debug link=static architecture=x86 --without-mpi --without-graph_parallel --without-type_erasure --without-stacktrace` it compiles (I even specify only filesystem and program_options in my CMakeLists.txt).

I see then there's a new version of Hunter, but, since it downloads the 1.83 version, it still fails to compile

CXX Compiler:
` [cmake] -- The CXX compiler identification is MSVC 19.40.33814.0 `

**Comment by john-forrest** [comment]
I believe this needs Boost 1.86.0 to fix. I'm currently trying to do this but even then there are problems - there is a required patch required for Boost 1.86.0 to install (should be fixed on 1.87.* :)) and I an finding the FindBoost in hunter get's confused and can't link Boost::filesystem on Windows - seems that both 32-bit and 64-bit libraries are installed, possibly to do with the patch, and cmake can't cope. I am getting around the Boost::filesystem issue by generally converting our codebase to use std::filesystem. (Unless you are supporting C++14, which is now the minimum version for Boost anyway,  or perhaps targeting a system that has issues, I doubt there are any advantages of using Boost::filesystem except perhaps not having to change many lines of existing code....)

**Comment by NeroBurner** [comment]
should be fixed with https://github.com/cpp-pm/hunter/pull/775 (with newer 1.86.0 Boost version)
